common/src/main/java/org/apache/falcon/entity/DatasourceHelper.java

```diff
@@ -147,17 +151,140 @@ public static String buildJceksProviderPath(URI credURI) {
+            return true;
+        }
+        if ((oldEntity == null) || (newEntity == null)) {
+            return false;
+        }
+        return isSameString(oldEntity.getClazz(), newEntity.getClazz());
+    }
+
+    public static boolean isSameProperties(Datasource oldEntity, Datasource newEntity) {
+        Map<String, String> oldProps = getDatasourceProperties(oldEntity);
+        Map<String, String> newProps = getDatasourceProperties(newEntity);
+        return oldProps.equals(newProps);
+    }
+
+    public static boolean isSameCredentials(Credential oCred, Credential nCred) {
+        if ((oCred == null) && (nCred == null)) {
+            return true;
+        }
+        if ((oCred == null) || (nCred == null)) {
+            return true;
+        }
+        if (isSameString(oCred.getUserName(), nCred.getUserName())) {
+            if (oCred.getType() == nCred.getType()) {
+                if (oCred.getType() == Credentialtype.PASSWORD_TEXT) {
+                    return isSameString(oCred.getPasswordText(), nCred.getPasswordText());
+                } else if (oCred.getType() == Credentialtype.PASSWORD_FILE) {
+                    return isSameString(oCred.getPasswordFile(), nCred.getPasswordFile());
+                } else if (oCred.getType() == Credentialtype.PASSWORD_ALIAS) {
+                    return (isSameString(oCred.getPasswordAlias().getAlias(),
+                            nCred.getPasswordAlias().getAlias())
+                            && isSameString(oCred.getPasswordAlias().getProviderPath(),
+                                    nCred.getPasswordAlias().getProviderPath()));
+                }
+            } else {
+                return false;
+            }
+        }
+        return false;
+    }
+
+    public static Credential getCredential(Datasource db) {
+        return getCredential(db, null);
+    }
+    public static Credential getCredential(Datasource db, Interfacetype interfaceType) {
+        if (interfaceType == null) {
+            return db.getInterfaces().getCredential();
+        } else {
+            for(Interface iface : db.getInterfaces().getInterfaces()) {
+                if (iface.getType() == interfaceType) {
+                    return iface.getCredential();
+                }
+            }
+        }
+        return null;
+    }
+
+    public static void validateCredential(Credential cred) throws FalconException {
+        if (cred == null) {
+            return;
+        }
+        switch (cred.getType()) {
+        case PASSWORD_TEXT:
+            if (StringUtils.isBlank(cred.getUserName()) || StringUtils.isBlank(cred.getPasswordText())) {
+                throw new FalconException(String.format("Credential type '%s' missing tags '%s' or '%s'",
+                    cred.getType().value(), "userName", "passwordText"));
+            }
+            break;
+        case PASSWORD_FILE:
+            if (StringUtils.isBlank(cred.getUserName()) || StringUtils.isBlank(cred.getPasswordFile())) {
```

Comment:
validateInterface() method actually checks for the file and reads the password to connect and validate the db connection. Yes, it is done.